### ISSUE TITLE
addPrimaryKey no longer exists

### ISSUE THREAD

**Issue by thinkingmedia** [cakephp/docs#3872]
In the section "Tips and Tricks"

http://book.cakephp.org/3.0/en/migrations.html#tips-and-tricks

It has this code sample:

```
<?php
use Migrations\AbstractMigration;

class CreateProductsTable extends AbstractMigration
{

    public $autoId = false;

    public function up()
    {
        $table = $this->table('products');
        $table
            ->addColumn('id', 'integer', [
                'autoIncrement' => true,
                'limit' => 11
            ])
            ->addPrimaryKey('id')
            ->addColumn('name', 'string')
            ->addColumn('description', 'text')
            ->create();
    }
}
```

I just tried it with CakePHP 3.2.6 and `addPrimaryKey` method does not exist on the table object.

**Comment by thinkingmedia** [comment]
It looks like this might be a table `$options` setting only now.

http://docs.phinx.org/en/latest/migrations.html#the-table-object

**Comment by markstory** [comment]
Would you be able to do a pull request to correct the book?

**Comment by thinkingmedia** [comment]
@markstory Yes, I will do that.

Phinx actually isn't very in-depth about primary keys for the docs. So give me a bit of time to figure this out.

**Comment by markstory** [comment]
@thinkingmedia Take your time, and thanks :smile:

**Comment by HavokInspiration** [comment]
I'll have to check but it is a method implemented in our own version of the Table object : 

https://github.com/cakephp/migrations/blob/master/src/Table.php#L30-L40

So if you use the migrations plugin ``AbstractMigration`` class (and not phinx one), the method will exist.

**Comment by thinkingmedia** [comment]
@HavokInspiration I am not seeing that class in my migrations. Looks to me like Phinx's table class is being used instead.

Don't waste time verifying this. I will fork and investigate when I have time.

**Comment by HavokInspiration** [comment]
It will depend on two things :

- the version of the migrations plugin you are using (this was implemented in migrations 1.3 if I remember correctly)
- what class your migrations classes are subclass of (``Migrations\AbstractMigration`` or ``Phinx\Migration\AbstractMigration``. If it is the second one, it won't work).

If you give me a gist of one of your migration class and the version of the plugin, I can check.
